mypy/checkexpr.py

```diff
@@ -356,10 +356,20 @@ def visit_call_expr_inner(self, e: CallExpr, allow_none_return: bool = False) ->
                     assert e.callee.fullname
                     callee_module = e.callee.fullname.rpartition(".")[0]
                 elif isinstance(e.callee, MemberExpr) and isinstance(e.callee.expr, NameExpr):
-                    assert e.callee.expr.fullname
-                    callee_module = e.callee.expr.fullname.rpartition(".")[0]
+                    if e.callee.expr.fullname:
+                        callee_module = e.callee.expr.fullname.rpartition(".")[0]
+                    elif e.callee.name == "__init_subclass__":
+                        t = get_proper_type(callee_type.bound_args[0])
+                        assert isinstance(t, TypeType)
+                        assert isinstance(t.item, Instance)
+                        assert isinstance(t.item.type, TypeInfo)
+                        callee_module = t.item.type.module_name
+                    else:
+                        # TODO: test time error
```

Comment:
what this means

Reply:
if this code gets reached under test it should have an error, but not in prod I think.